src/plugins/intel_gna/src/gna_graph_compiler.cpp

```diff
@@ -2404,22 +2404,15 @@ void GNAGraphCompiler::connectOutput(InferenceEngine::CNNLayerPtr layer, void* p
     log::debug() << "Connecting output " << layer->name << " ...\n";
     // in case of Memory Layer it's input allocated in meminput layer
     if (layer->outData.size() == 1) {
-        for (int j = 0; j != static_cast<int>(getInputTo(layer->outData.front()).size()); j++) {
-            auto isNonFunctional = [](CNNLayerPtr l) {
-                return LayerInfo(l).isNonFunctional();
-            };
-
-            if (!CNNNetHasNextLayerSkipCertain(layer, 0, j, isNonFunctional)) {
-                continue;
-            }
-            auto nextLayer = CNNNetGetNextLayerSkipCertain(layer, 0, j, isNonFunctional);
+        auto isNonFunctional = [](CNNLayerPtr l) {
```

Comment:
I have found around 17 places in the code where such lambda is created, maybe it would be good to create such method?

![image](https://github.com/openvinotoolkit/openvino/assets/106798706/2d1a697a-f742-4f01-8289-33555a6c6a74)

Reply:
I agree, but I'm wondering if it makes sense to refactor at this moment, especially since we're about to introduce ngraph API here soon.